project/src/main/kotlin/org/ionproject/core/accessControl/PDP.kt

```diff
@@ -0,0 +1,104 @@
+package org.ionproject.core.accessControl
+
+import org.ionproject.core.accessControl.pap.sql.AuthRepoImpl
+import org.ionproject.core.accessControl.pap.entities.PolicyEntity
+import org.ionproject.core.accessControl.pap.entities.TokenEntity
+import org.ionproject.core.common.customExceptions.ForbiddenActionException
+import org.ionproject.core.common.customExceptions.UnauthenticatedUserException
+import org.ionproject.core.common.interceptors.LoggerInterceptor
+import org.ionproject.core.common.interceptors.Request
+import org.ionproject.core.common.transaction.DataSourceHolder
+import org.ionproject.core.common.transaction.TransactionManagerImpl
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+private val logger = LoggerFactory.getLogger(LoggerInterceptor::class.java)
+
+@Component
+class PDP {
+  companion object {
```

Comment:
Why is everything on the companion object? This should be a regular component, injected into the components that require it.